src/backend/distributed/metadata/metadata_utility.c

```diff
@@ -614,7 +619,15 @@ GenerateSizeQueryOnMultiplePlacements(List *shardIntervalList, char *sizeQuery)
 		char *shardQualifiedName = quote_qualified_identifier(schemaName, shardName);
 		char *quotedShardName = quote_literal_cstr(shardQualifiedName);
 
-		appendStringInfo(selectQuery, sizeQuery, quotedShardName);
+		if (skipPartitions && PartitionedTable(shardInterval->relationId))
+		{
+			appendStringInfo(selectQuery, partitionedSizeFunction, quotedShardName);
+		}
+		else
+		{
+			appendStringInfo(selectQuery, sizeQuery, quotedShardName);
+		}
+
```

Comment:
Ah, ok, figured it out! But it indicates the code is to complex.

The list we are passing in is now tightly coupled with what we should be passing into the boolean flag.

I guess this is to save on compute time, looking at what I believe to be the issue you are solving (#4740). Would it be too much compute to get the original list of colocated placements and perform the `PartitionTable` filter here on the `relationId`? How much time are we saving? and is it worth the complications in understanding the code?

Reply:
Agreed, that would definitely increase code readability. But the flamegraph on #4740 reveals that not only `GenerateSizeQueryOnMultiplePlacements` (38.9%), but also `ColocatedShardIntervalList` (22.2%) causes this slowness we're trying to get rid of.  The root cause is the for loop in `ColocatedShardIntervalList`, which iterates all the colocated tables, because the number of these tables could be huge because of the partitions. So reducing the iteration count there (by skipping the partitions), saves us a significant computing time. I prefer this way, because the performance gain there is not negligible. 

What do you think about this tradeoff? @thanodnl @onderkalaci